addons/stock/models/stock_orderpoint.py

```diff
@@ -335,24 +335,22 @@ def _get_orderpoint_action(self):
         orderpoints_removed = orderpoints._unlink_processed_orderpoints()
         orderpoints = orderpoints - orderpoints_removed
         to_refill = defaultdict(float)
-        all_product_ids = self._get_orderpoint_products()
+        all_product_ids = self._get_orderpoint_products().ids
         all_replenish_location_ids = self.env['stock.location'].search([('replenish_location', '=', True)])
         ploc_per_day = defaultdict(set)
-        product_loc_by_qty = {}
         # For each replenish location get products with negative virtual_available aka forecast
-        for loc in all_replenish_location_ids:
-            quantities = all_product_ids.with_context(location=loc.id).mapped('virtual_available')
-            for product, quantity in zip(all_product_ids, quantities):
-                if float_compare(quantity, 0, precision_rounding=product.uom_id.rounding) >= 0:
-                    continue
-                product_loc_by_qty[(product, loc)] = quantity
-
-        # group product by lead_days and location in order to read virtual_available
-        # in batch
-        for (product, loc), qty in product_loc_by_qty.items():
-            rules = product._get_rules_from_location(loc)
-            lead_days = rules.with_context(bypass_delay_description=True)._get_lead_days(product)[0]
-            ploc_per_day[(lead_days, loc)].add(product.id)
+        for chunk in map(self.env['product.product'].browse, split_every(5000, all_product_ids)):
```

Comment:
You could use split_every

Reply:
I'm using `split_every`, did you mean to use in a different way? :)

Reply:
Oups my bad. Not it's okay this way

Reply:
Just maybe rename chunk in products, it would be easier to read

Reply:
Done